code/modules/power/cable.dm

```diff
@@ -448,21 +447,23 @@ GLOBAL_LIST_INIT(cable_coil_recipes, list(new/datum/stack_recipe("cable restrain
 		return FALSE
 	return TRUE
 
-GLOBAL_LIST(cable_radial_layer_list)
-
 /obj/item/stack/cable_coil/attack_self(mob/living/user)
 	if(!user)
 		return
-	if(!GLOB.cable_radial_layer_list)
-		GLOB.cable_radial_layer_list = list(
-		"Layer 1" = image(icon = 'icons/mob/radial.dmi', icon_state = "coil-red"),
-		"Layer 2" = image(icon = 'icons/mob/radial.dmi', icon_state = "coil-yellow"),
-		"Layer 3" = image(icon = 'icons/mob/radial.dmi', icon_state = "coil-blue"),
-		"Multilayer cable hub" = image(icon = 'icons/obj/power.dmi', icon_state = "cable_bridge"),
-		"Multi Z layer cable hub" = image(icon = 'icons/obj/power.dmi', icon_state = "cablerelay-broken-cable"),
-		"Cable crafting" = image(icon = 'icons/obj/items_and_weapons.dmi', icon_state = "cuff")
-		)
-	var/layer_result = show_radial_menu(user, src, GLOB.cable_radial_layer_list, custom_check = CALLBACK(src, .proc/check_menu, user), require_near = TRUE, tooltips = TRUE)
+
+	var/image/restraints_icon = image(icon = 'icons/obj/items_and_weapons.dmi', icon_state = "cuff")
+	restraints_icon.maptext = "<span class='maptext' [amount >= CABLE_RESTRAINTS_COST ? "" : "style='color: red'"]>[CABLE_RESTRAINTS_COST]</span>"
+
+	var/list/radial_menu = list(
+	"Layer 1" = image(icon = 'icons/mob/radial.dmi', icon_state = "coil-red"),
+	"Layer 2" = image(icon = 'icons/mob/radial.dmi', icon_state = "coil-yellow"),
+	"Layer 3" = image(icon = 'icons/mob/radial.dmi', icon_state = "coil-blue"),
+	"Multilayer cable hub" = image(icon = 'icons/obj/power.dmi', icon_state = "cable_bridge"),
+	"Multi Z layer cable hub" = image(icon = 'icons/obj/power.dmi', icon_state = "cablerelay-broken-cable"),
+	"Cable restraints" = restraints_icon
+	)
+
+	var/layer_result = show_radial_menu(user, src, radial_menu, custom_check = CALLBACK(src, .proc/check_menu, user), require_near = TRUE, tooltips = TRUE)
```

Comment:
You remove `GLOB.cable_radial_layer_list` and now create this list every time when someone open this radial menu,
i dont really know how this affects performance but i think using one list instead of creating new every time use less resurces

Reply:
Removing the global list is necessary so that the restraints icon can tell you if you have enough or not. If there is a noticable performance loss, then the constant options could be made once, with the list being copied over and appended to by the dynamic restraints one.